src/shared/components/copyDownloadControls/CopyDownloadControls.tsx

```diff
@@ -185,27 +182,25 @@ export class CopyDownloadControls extends React.Component<ICopyDownloadControlsP
         );
     }
 
-    public getText()
+    public getText(): string
     {
-        // if the download text is already there just return it
-        // else initiate an async download process
-        if (!this._downloadText) {
-            this.handleCopy();
-        }
-
-        return this._downloadText;
+        return this._copyText || "";
     }
 
     public handleCopy()
     {
-        this.initDownloadProcess(text => {
-            this.copyingData = true;
-        });
+        // async update of the copy text, if the copy text ends up to be different than the previous one,
+        // we show a modal to force a second copy action.
+        setTimeout(this.initCopyProcess(), 1);
```

Comment:
Fix for cBioPortal/cbioportal#3708. But, there is still an issue related to this behavior: cBioPortal/cbioportal#3709. Better to address that one in a separate PR.